src/ip_data.py

```diff
@@ -1,99 +1,72 @@
 import csv
 import random
 from datetime import datetime, timedelta
-from typing import Callable, Any, Iterator, Self, TypeVar
+from typing import Callable, Any, Iterator, TypeVar
 import base64
-
 import sqlite3
 
 T = TypeVar("T")
 
-NUM_ROWS = 100  # Number of rows to generate
-DDOS = 10
-START_TIME = datetime.now() - timedelta(days=1)
-END_TIME = datetime.now()
-
-ACCESS_SCHEMA = ["time", "ip", "end_point"]
-PEOPLE_SCHEMA = ["time", "person_name", "location_id"]
-SUBNET_SCHEMA = ["location_id", "location_name", "subnet_start", "subnet_end"]
-
-ENDPOINTS = [
-    "/users",
-    "/leaderboard",
-    "/food",
-    "/dinner",
-    "/register",
-    "/pay",
-]
-LOCATIONS = [
-    "bathroom",
-    "seminar-hall-1",
-    "seminar-hall-2",
-    "canteen-4-floor",
-    "canteen-5-floor",
-    "canteen-pixel-block",
-    "mrd-auditorium",
-]
-PERSON_NAMES = [
-    "potukuchi",
-    "naga",
-    "sriprad",
-    "reema",
-    "adhesh",
-    "gamblesh",
-    "aditya",
-    "sudhir",
-    "bevdesh",
-    "kiran",
-    "achyuth",
-    "suraj",
-    "abhinav",
-    "kaddapa",
-    "rowjee",
-    "🗿",
-]
-
-
-# Function to generate a mock IP address in the format 192.168.x.y
-def generate_ip() -> str:
-    x = random.randint(0, len(LOCATIONS) - 1)
+def generate_ip(locations_count: int) -> str:
+    x = random.randint(0, locations_count - 1)
     y = random.randint(0, 255)
     return f"{192}.{168}.{x}.{y}"
 
-
 def generate_time(start_time: datetime, end_time: datetime) -> datetime:
     delta = end_time - start_time
     random_seconds = random.randint(0, int(delta.total_seconds()))
     return start_time + timedelta(seconds=random_seconds)
 
-
 def interleave(iters: list[Iterator[T]]) -> Iterator[T]:
     while iters:
-        chosen = min(random.binomialvariate(1, 0.7), len(iters) - 1)
+        chosen = min(random.randint(0, len(iters) - 1), len(iters) - 1)
         it = iters[chosen]
         try:
             yield next(it)
         except StopIteration:
             iters.remove(it)
 
-
 def repeat(genrec: Callable[[], T], times: int) -> Iterator[T]:
-    for i in range(times):
+    for _ in range(times):
         yield genrec()
 
-
 class MockDB:
-    def __init__(self, dbid: int) -> None:
+    def __init__(self, dbid: int, num_rows: int = 100, ddos: int = 10, 
+                 start_time: datetime = None, end_time: datetime = None,
+                 access_schema=None, people_schema=None, subnet_schema=None,
+                 endpoints=None, locations=None, person_names=None) -> None:
```

Comment:
maybe don't have so many arguments in the constructor, how about you rewrite the code to use builder pattern, so that a user of `MockDB` can do `MockDB().with_rows(10000).with_ddos(4000)` please don't touch schemas, they are a different issue and trickier to handle